pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -150,6 +151,49 @@ func normalizeClusters(push *model.PushContext, proxy *model.Proxy, clusters []*
 	return out
 }
 
+// autoFillMTLSSettings adds service-level (m)TLS settings to the given destination rule, if applicable
+// (destination service can accept mTLS). The function returns the origin destination rule if it is already
+// has (service-level) TLS settings.
+// This function should be called for each service port, with the mTLS mode for that service + port.
+func autoFillMTLSSettings(config *model.Config, externalService bool,
+	destServiceMTLSMode authn_v1alpha1_applier.MutualTLSMode) *networking.DestinationRule {
+	destinationRule := &networking.DestinationRule{}
+	if config != nil {
+		destinationRule = config.Spec.(*networking.DestinationRule)
+		if destinationRule.TrafficPolicy != nil && destinationRule.TrafficPolicy.Tls != nil {
+			// Destination rule already has TLS settings, do nothing and return the DR as is.
+			// Note: technically, if DR has TLS settings in port-level or subset, we should returns the original as well.
+			// However, as port-level (or subset) will override the service-level, the code is simpler to add them regardless.
+			return destinationRule
+		}
+	}
```

Comment:
initialize the destination rule as part of else so that we dont waste allocating memory when config != nil

Reply:
Done